datadog/resource_datadog_security_monitoring_rule.go

```diff
@@ -0,0 +1,558 @@
+package datadog
+
+import (
+	"errors"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+
+	datadogV2 "github.com/DataDog/datadog-api-client-go/api/v2/datadog"
+)
+
+func resourceDatadogSecurityMonitoringRule() *schema.Resource {
+	return &schema.Resource{
+		Exists: resourceDatadogSecurityMonitoringRuleExists,
+		Create: resourceDatadogSecurityMonitoringRuleCreate,
+		Read:   resourceDatadogSecurityMonitoringRuleRead,
+		Update: resourceDatadogSecurityMonitoringRuleUpdate,
+		Delete: resourceDatadogSecurityMonitoringRuleDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: datadogSecurityMonitoringRuleSchema(),
+	}
+}
+
+func datadogSecurityMonitoringRuleSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"case": {
+			Type:        schema.TypeList,
+			Required:    true,
+			Description: "Cases for generating signals.",
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"name": {
+						Type:        schema.TypeString,
+						Optional:    true,
+						Description: "Name of the case.",
+					},
+					"condition": {
+						Type:        schema.TypeString,
+						Optional:    true,
+						Description: "A rule case contains logical operations (`>`,`>=`, `&&`, `||`) to determine if a signal should be generated based on the event counts in the previously defined queries.",
+					},
+					"notifications": {
+						Type:        schema.TypeList,
+						Optional:    true,
+						Description: "Notification targets for each rule case.",
+						Elem:        &schema.Schema{Type: schema.TypeString},
+					},
+					"status": {
+						Type:        schema.TypeString,
+						Required:    true,
+						Description: "Severity of the Security Signal.",
+					},
+				},
+			},
+		},
+
+		"enabled": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Description: "Whether the rule is enabled.",
+		},
+
+		"disabled": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Description: "Whether the rule is disabled.",
+		},
```

Comment:
Why do we need two fields?

Reply:
https://github.com/DataDog/terraform-provider-datadog/pull/763#discussion_r530557706
It is a limitation in the way to check for optional fields in terraform `d.GetOk()` will only return `ok = true` if a value is present and truthy. Having only an enabled field means that rules would need to be disabled by default. This would be consistent with [logs_custom_pipelines](https://registry.terraform.io/providers/DataDog/datadog/latest/docs/resources/logs_custom_pipeline) but it seems strange to me to have to specify `enabled: true` for each rule resource.

Reply:
Could we set the default value of enabled to true instead?

Reply:
Ah! I was trying to implement default in code instead of using the default in the resource definition. I'll do that!